components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -215,6 +216,37 @@ func (r *NotebookReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, err
 	}
 
+	// Check if annotations for the Notebook instance imply a requirement for notebook restart.
+	// This annotation is supposed to be added once a ConfigMap used in the notebook is added/updated.
+
+	annotations := instance.GetAnnotations()
+	notebookRestart, ok := annotations[AnnotationNotebookRestart]
+
+	if ok && notebookRestart == "true" {
+
+		// find the pod associated with the notebook instance
+		foundPod := &corev1.Pod{}
+		err = r.Get(ctx, types.NamespacedName{Name: instance.Name + "-0", Namespace: instance.Namespace}, foundPod)
+		if err != nil && apierrs.IsNotFound(err) {
+			log.Info(fmt.Sprintf("No Pods are currently running for Notebook Server: %s in namesace: %s.", instance.Name, instance.Namespace))
+		} else if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		// delete the pod associated with the notebook instance so it can be restarted
+		r.Delete(context.TODO(), foundPod)
```

Comment:
lets include the annotation deletion logic inside else, so that if deletion doesnt happens , notebook-controller doesnt happen..
```suggestion
		err = r.Delete(context.TODO(), foundPod)
		if err != nil {
			return ctrl.Result{}, err
		} else {
		
```
WDYT?

Reply:
This will be the correct way!